### PR TITLE
scrapy crawling data from codeforces

### PR DESCRIPTION
**Pull request by william1nguyen**

### CODE REVIEW

File: cohostforce-data/cohostforce/items.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Define here the models for your scraped items
+#
+# See documentation in:
+# https://docs.scrapy.org/en/latest/topics/items.html
+
+import scrapy
+
+
+class CohostforceItem(scrapy.Item):
```
Comment (coleslaw) [inline cohostforce-data/cohostforce/items.py]: not used?

File: cohostforce-data/cohostforce/middlewares.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+# See documentation in:
+# https://docs.scrapy.org/en/latest/topics/spider-middleware.html
+
+from scrapy import signals
+
+# useful for handling different item types with a single interface
+from itemadapter import is_item, ItemAdapter
+
+
+class CohostforceSpiderMiddleware:
```
Comment (coleslaw) [inline cohostforce-data/cohostforce/middlewares.py]: Rename `CohostForceSpiderMildleware`

**Review by coleslaw — APPROVED** [review]
Looks great !

File: cohostforce-data/cohostforce/spiders/problem.py [hunk 3]
```diff
@@ -10,8 +10,8 @@ class ProblemSpider(scrapy.Spider):
     start_urls = []
 
     def start_requests(self):
-        pages_number = 2
-        for page_id in range(1, pages_number):
+        pages_number = 86
+        for page_id in range(85, pages_number):
```
Comment (luzivaltic) [inline cohostforce-data/cohostforce/spiders/problem.py]: why need page_id ?

**Pull request merged**